### ISSUE TITLE
airframe-http-finagle: Support custom codec to encode responses

### ISSUE THREAD

**Issue by Mura-Mi** [wvlet/airframe#791]
airframe-http-finagle fails to encode return value if `MessageCodecFactory.defaultFactory.withObjectMapCodec` can't prepare the codec for endpoint's return type. This is because developer can't add custom codecs in `FinagleResponseHandler#mapCodecFactory`.

https://github.com/wvlet/airframe/blob/82c92be0895e3b5120b24f4aa57f294e9d60c356/airframe-http-finagle/src/main/scala/wvlet/airframe/http/finagle/FinagleResponseHandler.scala#L33-L34

For example: 
```scala
case class User(name: String,
                birthday: java.time.LocalDate // mapCodecFactory doesn't know Codec[LocalDate]
)

@Endpoint trait SomeEndpoint {
  @Endpoint(path="nomatter") def get(): User // mapCodecFactory.of[User].toMsgPack fails in FinagleResponseHandler
}
```

Would you consider introducing extension points? Following is my own idea for just example.
* Make mapCodecFactory protected. Developer can extend `FinagleResponseHandler` to override this value and add it to DI design.
* Prepare an extension point (`protected def` or class value) to declare custom codecs to be used in `FinagleResponseHandler`. Developers can add custom codecs by extending this class or creating non default instance.

**Comment by xerial** [comment]
- I think providing a config for using custom codec is better than overriding some server definition classes.
- Supporting LocalDate in airframe-codec is an option, but we generally do not recommend using LocalDate as it doesn't have any time zone information and can cause a lot of troubles. Using ZonedDateTime or Instant is better.

**Comment by xerial** [comment]
A support is implemented in #797

**Comment by xerial** [comment]
Added in 19.11.0
